In [11]:
#Project 1 Task 2, Linear Regression with Multiple Variables
#Christian Johnston, CPSC 4383, Fall 2023

#This method is very similar to the single variable linear regression data model, but with multiple variables for x.
#I will be using a different reference though: https://www.w3schools.com/python/python_ml_multiple_regression.asp

#first I am going to set up my tools for the project: pandas, and scikit learn.

import pandas as pd
from sklearn import linear_model

#using pandas i extract the data into my variables.

data = pd.read_csv(r'C:\Users\Christian\Desktop\School Stuff\Fall 2023\AI\course content\ex1data2.csv')

#now that we have the data, we can extract the columns we need for our linear regression fit.

X = pd.DataFrame(data, columns=['Size of the House', 'Number of Bedrooms'])
y = pd.DataFrame(data, columns=['Price'])


#Here i construct the model for linear regression using the linear_model method in scikit learn.
model = linear_model.LinearRegression()
model.fit(X,y)

#now that our model is constructed, lets make a prediction to test it against the data.
# lets say we want a house that is 2000 sqft with 3 bedrooms.

print(model.predict([[2000, 3]]))

[[341805.20024107]]


C:\Users\Christian\Desktop\School Stuff\Fall 2023\AI\Anaconda\Lib\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


Here we see the predicted values for a 2000sqft house with 3 bedrooms.  The model uses ordinary least squares but this time uses two different data columns to calculate the best fit.  The values from my test run gave me 341805.20024107 which sounds about right considering my personal observation of the data.